graylog2-web-interface/src/views/components/sidebar/ContentColumn.jsx

```diff
@@ -0,0 +1,162 @@
+import { type ViewMetaData as ViewMetadata } from 'views/stores/ViewMetadataStore';
+
+import ViewTypeLabel from 'views/components/ViewTypeLabel';
+import ViewTypeContext from 'views/components/contexts/ViewTypeContext';
+import { IconButton } from 'components/common';
+
+type Props = {
+  children: React.Node,
+  closeSidebar: () => void,
+  searchPageLayout: ?SearchPageLayout,
+  sectionTitle: string,
+  viewMetadata: ViewMetadata,
+};
+
+export const Container: StyledComponent<{ sidebarIsInline: boolean }, ThemeInterface, HTMLDivElement> = styled.div(({ theme, sidebarIsInline }) => `
+  position: ${sidebarIsInline ? 'static' : 'fixed'}
+  display: grid;
+  display: -ms-grid;
+  grid-templage-columns: 1fr;
+  grid-template-rows: auto 1fr;
+  -ms-grid-columns: 1fr;
+  -ms-grid-rows: auto 1fr;
+  top: ${sidebarIsInline ? 0 : '50px'};
+  left: ${sidebarIsInline ? 0 : '50px'};
+
+  width: 270px;
+  height:  ${sidebarIsInline ? '100%' : 'calc(100% - 50px)'};;
+  padding: 5px 15px 15px 15px;
+
+  color: ${theme.colors.global.textDefault};
+  background: ${theme.colors.global.contentBackground};
+  border-right: 1px solid ${theme.colors.gray[80]};
+  
+  overflow-y: auto;
+  z-index: 3;
+
+  > *:nth-child(1) {
+    grid-column: 1;
+    -ms-grid-column: 1;
+    grid-row: 1;
+    -ms-grid-row: 1;
+  }
+
+  > *:nth-child(2) {
+    grid-column: 1;
+    -ms-grid-column: 1;
+    grid-row: 2;
+    -ms-grid-row: 2;
+  }
+`);
+
+const Header: StyledComponent<{}, void, HTMLDivElement> = styled.div`
+  height: 35px;
+  display: grid;
+  display: -ms-grid;
+  grid-template-columns: 1fr auto;
+  -ms-grid-columns: 1fr auto;
+
+  > *:nth-child(1) {
+    grid-column: 1;
+    -ms-grid-column: 1;
+  }
+
+  > *:nth-child(2) {
+    grid-column: 2;
+    -ms-grid-column: 2;
+  }
+`;
+
+const Title = styled.h1`
+  color: inherit;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  font-size: 24px;
+  cursor: pointer;
+`;
+
+const OverlayToggle: StyledComponent<{ sidebarIsInline: boolean }, ThemeInterface, HTMLDivElement> = styled.div(({ theme, sidebarIsInline }) => `
+  > * {
+    font-size: 18px;
+    color: ${sidebarIsInline ? theme.colors.gray[30] : theme.colors.variant.info};
+  }
+`);
+
+const HorizontalRule = styled.hr`
+  margin: 5px 0 10px 0;
+`;
+
+const SectionTitle = styled.h2`
+  margin-bottom: 10px;
+`;
+
+const CenterVertical = styled.div`
+  display: inline-grid;
+  display: -ms-inline-grid;
+  align-content: center;
+`;
+
+const toggleSidebarPinning = (searchPageLayout) => {
+  if (!searchPageLayout) {
+    return;
+  }
+  const { setConfig, config } = searchPageLayout;
+  const sidebarIsInline = config?.sidebar.isInline;
+  const newLayoutConfig = {
+    ...config,
+    sidebar: { isInline: !sidebarIsInline },
+  };
+  setConfig(newLayoutConfig);
+};
+
+const sidebarTitle = (viewMetadata: ViewMetadata, viewType: ?ViewType) => {
+  if (!viewMetadata.id) {
```

Comment:
We should use `isNew` from `ViewStore` instead.

Reply:
Using `isNew` here would lead to a slightly different behaviour. We would never display the title `Untitled Dashboard`. Right now this title is being display when a user creates a new dashboard (`/dashboards/new`).

Do you prefer to display the title for the default search in this case as well?

Reply:
Imo these are two separate concerns. One is to find out if we have a new, unsaved search/dashboard, the other is to format the title. In my opinion, we should use `isNew` to determine the first and use `viewType` to format a title according to the type.

Reply:
Thanks for the helpful feedback, I've updated the code. I also change the default title for unsaved views to `Unsaved ...` in the same step.